index/scorch/persister.go

```diff
@@ -212,7 +212,8 @@ OUTER:
 		case s.introducerNotifier <- w:
 		}
 
-		s.removeOldData() // might as well cleanup while waiting
+		forceCompact := s.isForceCompacted(ourSnapshot)
```

Comment:
somehow... this feels racey?  not sure...

...but thinking as I'm reading this codepath... that the force-compact happened long ago...

and it's somehow affecting how you want to cleanup old files right now?

wondering if we didn't change the removeOldData() behavior, would the other parts of the force-compact proposal still just work?